mmaction/core/optimizer/trn_optimizer_constructor.py

```diff
@@ -0,0 +1,109 @@
+import torch
+from mmcv.runner import OPTIMIZER_BUILDERS, DefaultOptimizerConstructor
+from mmcv.utils import SyncBatchNorm, _BatchNorm, _ConvNd
+
+
+@OPTIMIZER_BUILDERS.register_module()
+class TSMOptimizerConstructor(DefaultOptimizerConstructor):
```

Comment:
Guess it should be TRNOptimizerConstructor?

Reply:
emmm, I forgot to delete the file. After I copied tsm_optimizer_consturtor.py and were going to modify it, I found that I can directly use TSM's.

Reply:
That's OK. So does TRN need a custom optimizer?

Reply:
Yes :p